workflows/pipe-common/scripts/blocked_users_cleanup.py

```diff
@@ -0,0 +1,235 @@
+        except KeyboardInterrupt:
+            logger.warning('Interrupted.')
+            raise
+        except Exception:
+            logger.warning('Running instances cleanup has failed.', trace=True)
+    logger.info('Finishing running instances cleanup...')
+
+def _cleanup_tools(api, logger, users, dry_run):
+    exp_days = os.getenv('CP_CAP_TOOL_EXP_DAYS')
+    if exp_days is None:
+        return
+    exp_days = int(exp_days)
+    tools_users = _get_cleanup_users(users, exp_days)
+    if len(tools_users) == 0:
+        return
+    logger.info('Loading tools...')
+    registries = api.docker_registry_load_all()
+    tools = _get_tools(registries)
+    logger.info('Processing tools...')
+    for tool in tools:
+        try:
+            if tool['owner'] in tools_users:
+                permissions = api.get_permissions(tool['id'], 'TOOL')
+                if permissions is None:
+                    if not dry_run:
+                        api.delete_tool(tool['image'])
+                    logger.info('Tool {} deleted.'.format(tool['id']))
+                else:
+                    logger.info('Tool {} was not deleted because it is shared with other users.'.format(tool['id']))
+        except KeyboardInterrupt:
+            logger.warning('Interrupted.')
+            raise
+        except Exception:
+            logger.warning('Tools cleanup has failed.', trace=True)
+    logger.info('Finishing tools cleanup...')
+
+def _cleanup_storages(api, logger, users, dry_run):
+    logger.info('Loading data storages...')
+    storages = api.data_storage_load_all()
+    logger.info('Processing data storages...')
+    file_storages = _get_file_data_storages(storages)
+    for storage in file_storages:
+        try:
+            user = next(filter(lambda u: u['userName'] == storage['owner'], users))
+            if _is_home_storage(users, storage['id']):
+                _cleanup_home_datastorage(api, logger, storage, user, dry_run)
+            else:
+                _cleanup_general_datastorage(api, logger, storage, user, dry_run)
+        except KeyboardInterrupt:
+            logger.warning('Interrupted.')
+            raise
+        except Exception:
+            logger.warning('Data storage cleanup has failed.', trace=True)
+    logger.info('Finishing data storages cleanup...')
+
+def _is_home_storage(users, storage_id):
+    return storage_id in [u['defaultStorageId'] for u in users]
+
+def _replace_datastorage_content(logger, source, target):
+    executor = LocalExecutor()
+    executor = LoggingExecutor(logger=logger, inner=executor)
+    executor.execute('{} "{}" "{}" --recursive --force --quiet'.format(PIPE_STORAGE_CP, source, target))
+
+def _cleanup_home_datastorage(api, logger, storage, user, dry_run):
+    exp_days = os.getenv('CP_CAP_DATASTORE_DEFAULT_EXP_DAYS')
+    if exp_days is None:
+        return
+    if not _clean_up(user, exp_days):
+        return
+    storage_items = api.load_datastorage_items(storage['id'])
```

Comment:
Let's check permissions as well + please check if storage is a parent of any other storage. E.g. storage1 with path `/path/folder`, and storage2 with `/path/folder/inner`, then storage2 is a shared copy of storage1